christin/index.js

```diff
@@ -0,0 +1,237 @@
+/* 1. Check if a number is within a given range. Write a program that checks if a number is within the range of an object's min and max properties. Assume min <= max is always true.
+Examples:
+4, { min: 0, max: 5 }) ➞ true
+4, { min: 4, max: 5 }) ➞ true
+4, { min: 6, max: 10 }) ➞ false
+5, { min: 5, max: 5 }) ➞ true*/
+
+const range = {
+  inRange: function (number, min, max) {
+    if (number > min && number < max) {
+      return true;
+    } else {
+      return false;
+    }
+  },
+};
+
+console.log("in range:", range.inRange(4, 0, 5)); // true
+console.log("in range:", range.inRange(4, 6, 10)); // false
+
+// 2. Scrabble. Write a program that, given an array of scrabble tiles, counts the maximum score that a player can earn from the tiles in their hand. Example: [ { tile: "N", score: 1 }, { tile: "K", score: 5 }, { tile: "Z", score: 10 }, { tile: "X", score: 8 }, { tile: "D", score: 2 }, { tile: "A", score: 1 }, { tile: "E", score: 1 } ]
+
+//The player's maximum score: 1 + 5 + 10 + 8 + 2 + 1 + 1 = 28
+
+const scrabbleTiles = [
+  { tile: "N", score: 1 },
+  { tile: "K", score: 5 },
+  { tile: "Z", score: 10 },
+  { tile: "X", score: 8 },
+  { tile: "D", score: 2 },
+  { tile: "A", score: 1 },
+  { tile: "E", score: 1 },
+];
+
+const maxScoreCounter = (array) => {
+  let sum = 0;
+  for (let i = 0; i < array.length; i++) {
+    sum += array[i].score;
+  }
+  return sum;
+};
+
+console.log(maxScoreCounter(scrabbleTiles));
+
+//! 3. Is it an empty object? Write a program that returns true if an object is empty, and false if otherwise.
+
+//Examples:
+//{ } ➞ true
+//{ a: 1 } ➞ false
+
+const empty = {};
+const value = {
+  val1: 1,
+  val2: 2,
+};
+
+function checkObject(object) {
+  let objectToArray = Object.entries(object);
+  if (objectToArray.length <= 0) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+console.log("checkObject", checkObject(value));
+
+/* 4. Counting Letters. Create a function that counts the number of occurrences of each letter in a string. Return an object with key pair values of letters and the number of occurrences for each letter.
+Example:
+countLetters("tree") ➞ {t: 1, r: 1, e: 2}*/
+
+let ourString = "thats a string";
+
+const countLetters = (string) => {
```

Comment:
Task 4 ==> When we run the code we'll get this output { t: 1, h: 1, a: 1, count: NaN, s: 1, ' ': 1, r: 1, i: 1, n: 1, g: 1 }
So here is some tips ==> 
// First option ==> within for loop we have to store string.length instead of ourString.length
Then in if statement (mapOfOccurrences[string[i]])  the logic should be like this mapOfOccurrences[string[i]]++;
instead of maoOfOccurrences.count +=1


- before const mapOfOccurrences you can make a constant and use split method to split the characters like for example const strToArr = string.split("")
- within if statement we could use hasOwnProperty Object method. The hasOwnProperty() method returns a boolean indicating whether the object has the specified property as its own property (as opposed to inheriting it).
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
means if(mapOfOccurrences.hasOwnProperty(strToArr[i])) then mapOfOccurrences[strToArr[i] ++
otherwise mapOfOccurrences[strToArr[i]]

Reply:
modified number 4